add python solution for icy perimeter (895)

jjliewie opened the pull request:
_If any of the below doesn't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code. -> okay for this one I did test my code and I swear that it used to work perfectly but when I just ran it the last test case tled... On the usaco site it does say "Note also that we have recently changed grading servers, and since our new servers run at different speeds from the servers used during older contests, timing results for older contest problems may be slightly off until we manage to re-calibrate everything properly." Time complexity wise it should work is my point and it did work before. On other sites (like on the Chinese site and the Korean site -> those are the only two I know) it works. 
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which includes the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making them.

SansPapyrus683 commented on solutions/usaco-895.mdx:
```diff
@@ -169,4 +169,76 @@ int main() {
+import sys
+from collections import deque
+
+sys.stdin = open("perimeter.in", "r")
+sys.stdout = open("perimeter.out", "w")
+
```
.

SansPapyrus683 commented on solutions/usaco-895.mdx:
```diff
@@ -169,4 +169,76 @@ int main() {
+min_peri = float('inf')
+
+# check if out of bounds
+def out(a, b, l):
```
type annotations

SansPapyrus683 replied:
not resolved?

SansPapyrus683 commented on solutions/usaco-895.mdx:
```diff
@@ -169,4 +169,76 @@ int main() {
+	area, peri = 1, 0
+
+	# possible directions
+	directions = [
```
rename to DIRECTIONS

SansPapyrus683 commented on solutions/usaco-895.mdx:
```diff
@@ -169,4 +169,76 @@ int main() {
+		return True
+	return False
+
+# bfs
+def bfs(x, y):
```
name this something actually informative like `area_and_perimeter`

SansPapyrus683 commented:
if this TLEs it's fine just add a warning tag saying smth like "python slow af so this thing can't AC"

ryanchou-dev reviewed (approved):
aside from sans' suggestions, looks good

SansPapyrus683 commented on solutions/usaco-895.mdx:
```diff
@@ -169,4 +169,72 @@ int main() {
+		return True
+	return False
+
+# bfs
```
what's this `bfs`-ing?

jjliewie replied:
breadth first search

SansPapyrus683 replied:
ok uh the function name is fine now, just delete this

jjliewie replied:
okay

jjliewie commented:
> if this TLEs it's fine just add a warning tag saying smth like "python slow af so this thing can't AC"

done.. . i think addressed all the suggestions?
